enhancements/two-nodes-openshift/2no.md

```diff
@@ -0,0 +1,618 @@
+       * Mitigation (Phase 1): manual intervention (possibly a script)  in case admin can guarantee Node2 is down, which will grant Node1 quorum and restore cluster limited (none HA) functionality.
+       * Mitigation (Phase 2): limited automatic intervention for some use cases: for example Node1 will gain quorum only if Node2 can be verified to be down by successfully querying its BMC status.
+5. Kubelet Failure
+   1. Pacemaker’s monitoring detects the failure
+   2. Pacemaker restarts kubelet
+   3. Stop failure is optionally escalated to a node failure (fencing)
+   4. Start failure defaults to leaving the service offline
+6. Etcd Failure
+   1. Pacemaker’s monitoring detects the failure
+   2. Pacemaker either demotes etcd so it can resync, or restarts and promotes etcd
+   3. Stop failure is optionally escalated to a node failure (fencing)
+   4. Start failure defaults to leaving the service offline
+
+
+### API Extensions
+
+API Extensions are CRDs, admission and conversion webhooks, aggregated API servers,
+and finalizers, i.e. those mechanisms that change the OCP API surface and behaviour.
+
+- Name the API extensions this enhancement adds or modifies.
+- Does this enhancement modify the behaviour of existing resources, especially those owned
+  by other parties than the authoring team (including upstream resources), and, if yes, how?
+  Please add those other parties as reviewers to the enhancement.
+
+  Examples:
+  - Adds a finalizer to namespaces. Namespace cannot be deleted without our controller running.
+  - Restricts the label format for objects to X.
+  - Defaults field Y on object kind Z.
+
+Fill in the operational impact of these API Extensions in the "Operational Aspects
+of API Extensions" section.
+
+### Topology Considerations
+
+2NO represents a new topology, and is not appropriate for use with HyperShift, SNO, or MicroShift
+
+#### Hypershift / Hosted Control Planes
+
+Are there any unique considerations for making this change work with
+Hypershift?
+
+See https://github.com/openshift/enhancements/blob/e044f84e9b2bafa600e6c24e35d226463c2308a5/enhancements/multi-arch/heterogeneous-architecture-clusters.md?plain=1#L282
+
+How does it affect any of the components running in the
+management cluster? How does it affect any components running split
+between the management cluster and guest cluster?
+
+#### Standalone Clusters
+
+Is the change relevant for standalone clusters?
+
+#### Single-node Deployments or MicroShift
+
+How does this proposal affect the resource consumption of a
+single-node OpenShift deployment (SNO), CPU and memory?
+
+How does this proposal affect MicroShift? For example, if the proposal
+adds configuration options through API resources, should any of those
+behaviors also be exposed to MicroShift admins through the
+configuration file for MicroShift?
+
+### Implementation Details/Notes/Constraints
+
+#### Installation flow
```

Comment:
How is a 2NO installation identified in the install-config?
Does setting controlPlane.replicas = 2 mean 2NO? 
Are additional settings for the BMCs needed?

Reply:
>Are additional settings for the BMCs needed?

IIUC yes, we've touched that a bit in the previous meeting but we still need to discuss which setting are needed and how to add them.

>How is a 2NO installation identified in the install-config?
>Does setting controlPlane.replicas = 2 mean 2NO?

Those are great question but  IIUC they should probably be decided by the installer experts.
Feel free to discuss here/add to the meeting agenda.

Reply:
Resolved